### PR TITLE
Add max_age_days filter to search_issues

### LINKED ISSUE #155: [ENHANCEMENT]: Let MCP search_issues filter by how old the issue is
## Problem

Half the dataset is not really available.  An issue labeled `good first issue` that nobody touched in three years is not a good first issue, it is a label somebody forgot to remove.

`search_issues` can't do anything about that. It filters by language, comments, label and repo, and the two date fields we already collect are just along for the ride. They go into the CSV, they reach the MCP server, and then nothing reads them.

So when somebody asks their assistant for something to work on, they get a mix of live issues and issues from 2016, with no way to tell them apart or ask for only the recent ones.

## Desired behavior

`search_issues` takes a way to ask for recent issues only, and the assistant can use it. Something like `max_age_days=90` returning only what was opened in the last three months.

Whether it filters on `created_at`, on `updated_at`, or lets you choose, is the part worth thinking about. They answer different questions: `created_at` is how long the issue has existed, `updated_at` is whether anybody is still paying attention to it. Say in the comments which one you picked and why, either is fine as long as the tool description tells the model what it does.

## Thoughts & Additional Info

Where to look:

- `search_issues` takes the arguments and passes them down, `app/mcp_server.py`
- `filter_issues` does the actual filtering, `app/core/dataset.py`
- Both dates are already parsed and sitting in every issue dict

The docstring of the tool is what the model reads to decide when to call it, so it matters more than a normal docstring. `docs/mcp-server.md` documents every argument under "Available tools" and needs the new one.

CI fails under 100% coverage, so the tests go in the same PR. See TESTING.md if you haven't run the suite before.

Comment here to claim it and I will assign it to you.

### PR DESCRIPTION
**Pull request by SHEKHAR-Y**
## What does this PR do?
Filtering the issues on the basis of updated_at so that only issues which are active will be searched, also the max_age_days is inclusive of the boundary to ensure if it is set 0 it will return the issues opened or updated today.

Issues with empty updated_at are excluded when max_age_days is set, since we can't verify the recency of the issue and the whole point of this feature is filtering out the stale issues.

Added 4 new tests covering basic age filtering, inclusive boundary case,  empty updated_at exclusion and when max_age_days not passed.

Full suite passes, 100% coverage maintained. 

## Related Issue
Fixes #155

## Checklist

- [ x ] I read the [CONTRIBUTING.md](../CONTRIBUTING.md)
- [ x ] I ran the project locally and tested my changes
- [ x ] I verified my changes are working as expected
- [ x ] This PR description is written by me, not by AI

### CODE REVIEW

File: app/tests/test_mcp_server.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import asyncio
 import sys
 from unittest.mock import patch
+from datetime import date, timedelta
 
 import pytest
 
 from mcp.server.mcpserver import MCPServer
+from app.core.dataset import DatasetManager
```
Comment (drkrillo) [inline app/tests/test_mcp_server.py]: We should leave this import out since it's not used.

**Review by drkrillo — COMMENTED** [review]
Hi @SHEKHAR-Y !

PR is looking good. 

Left a minor comment in the code: In`app/tests/test_mcp_server.py`, `DatasetManager` is imported but never used. Push that one and we are ready to merge. Thanks for the back and forth on the issue, the `updated_at` call was yours and it was the right one.

File: app/tests/test_mcp_server.py (hunk 1 above)
Comment (SHEKHAR-Y) [reply app/tests/test_mcp_server.py]: Hi @drkrillo !
Pushed the requested changes. I’ve removed the unused DatasetManager import. Thanks for the review!

**Review by drkrillo — APPROVED** [review]

**Comment by drkrillo** [comment]
thank you @SHEKHAR-Y for the quick change! Merging.

Thanks for contributing to the repo :)
